### PR TITLE
Functional components

### PR DESCRIPTION
**Pull request by jordanoverbye**
There are currently a few components that do not hold any state, so I have just converted them into functions.

### CODE REVIEW

**Comment by j-f1** [comment]
Switching them to arrow functions will get rid of another level of indentation. What do you think?

**Comment by jordanoverbye** [comment]
Hey @j-f1, the reason I didn't use arrow functions is because on the reactjs.org website it seems every functional component is written with the standard function syntax, not an arrow function. I just assumed that the react team encourages this pattern - but if arrow functions would be better, then I am happy to make another commit.

**Comment by j-f1** [comment]
Ah, interested to hear what the team’s position is on that!

**Comment by jordanoverbye** [comment]
@j-f1 Me too. I've just pushed a new commit to resolve those conflict errors.

**Comment by gaearon** [comment]
Sorry, been making some changes and didn't have time to look into this.

If you do this on master and switch classes to Hooks where necessary that sounds cool

**Comment by willisplummer** [comment]
@gaearon just out of curiosity, why do you prefer regular named functions over arrow functions?

File: src/components/Layout.js [hunk 1]
```diff
@@ -1,136 +1,129 @@
...
-class Layout extends React.Component {
-  state = {
-    theme: null,
-  };
-  componentDidMount() {
-    this.setState({ theme: window.__theme });
+function Layout(props) {
+  const [theme, setTheme] = useState(null);
+
+  useEffect(() => {
```
Comment (gaearon) [inline src/components/Layout.js]: I think this should be a layout effect.

Comment (jordanoverbye) [reply src/components/Layout.js]: @gaearon I've just updated this in my latest commit

**Pull request closed without merging**